modules/tensor_mechanics/src/bcs/DisplacementAboutAxis.C

```diff
@@ -69,39 +70,46 @@ DisplacementAboutAxis::computeQpValue()
 ColumnMajorMatrix
 DisplacementAboutAxis::rotateAroundAxis(const ColumnMajorMatrix & p0, const Real angle)
 {
-  ColumnMajorMatrix rotz(4,4);
-  rotz(0,0) = cos(angle);
-  rotz(0,1) = -sin(angle);
-  rotz(0,2) = 0;
-  rotz(0,3) = 0;
-  rotz(1,0) = sin(angle);
-  rotz(1,1) = cos(angle);
-  rotz(1,2) = 0;
-  rotz(1,3) = 0;
-  rotz(2,0) = 0;
-  rotz(2,1) = 0;
-  rotz(2,2) = 1;
-  rotz(2,3) = 0;
-  rotz(3,0) = 0;
-  rotz(3,1) = 0;
-  rotz(3,2) = 0;
-  rotz(3,3) = 1;
-
-  ColumnMajorMatrix transform = _transformation_matrix_inv * rotz * _transformation_matrix;
+  ColumnMajorMatrix rotate_about_z(4,4);
+  rotate_about_z(0,0) = cos(angle);
+  rotate_about_z(0,1) = -sin(angle);
+  rotate_about_z(0,2) = 0;
+  rotate_about_z(0,3) = 0;
+  rotate_about_z(1,0) = sin(angle);
+  rotate_about_z(1,1) = cos(angle);
+  rotate_about_z(1,2) = 0;
+  rotate_about_z(1,3) = 0;
+  rotate_about_z(2,0) = 0;
+  rotate_about_z(2,1) = 0;
+  rotate_about_z(2,2) = 1;
+  rotate_about_z(2,3) = 0;
+  rotate_about_z(3,0) = 0;
+  rotate_about_z(3,1) = 0;
+  rotate_about_z(3,2) = 0;
+  rotate_about_z(3,3) = 1;
+
+  ColumnMajorMatrix transform = _transformation_matrix_inv * rotate_about_z * _transformation_matrix;
   return transform * p0;
 }
 
 void
+DisplacementAboutAxis::calculateUnitDirectionVector()
+{
+  Real magnitude = _axis_direction.norm();
+
+  // only perform the operation if the provided direction vector is not already a unit vector
+  if (std::abs(1.0 - magnitude) > 1.0e-15)
+    for (unsigned int i = 0; i < 3; ++i)
+      _axis_direction(i) = _axis_direction(i) / magnitude;
```

Comment:
This seems possible, there is no check in c'tor that would prevent it. Let's add that check there...